addons/website_slides/populate/slide_channel_partner.py

```diff
@@ -0,0 +1,62 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+from collections import defaultdict
+
+from odoo import models
+from odoo.tools import populate
+
+
+class SlideChannelPartner(models.Model):
+    _inherit = 'slide.channel.partner'
+    _populate_dependencies = ['res.partner', 'slide.channel']
+    # 50% more than partners, so supports 0 to N courses per partner.
+    _populate_sizes = {'small': 150, 'medium': 3_000, 'large': 150_000}
+
+    # Will ensure some courses have many partners
+    choices_weights = {}
```

Comment:
it's a bit annoying to have that as a class attribute, can't we use a variable inside of the method instead ?

Reply:
so you can just use lru_cache ? what do you think ?

(I don't like the small complexity it add))